src/libsyntax/ext/expand.rs

```diff
@@ -347,7 +347,7 @@ impl<'a, 'b> MacroExpander<'a, 'b> {
                     let derives = derives.entry(invoc.expansion_data.mark).or_insert_with(Vec::new);
 
                     for path in &traits {
-                        let mark = Mark::fresh(self.cx.current_expansion.mark);
```

Comment:
I was more worried about this causing breaking changes to `macro_rules` than breaking the unstable macros 2.0. But I also cannot come up with an example that would break due to this change. Maybe we should do a crater run?

Reply:
Yeah, that would be interesting since unstable crates are on crates.io too.